### PR TITLE
Make EventProcessor.compose() additive

### PR DESCRIPTION
**Pull request by davidwheeler123**
### Description

Two main changes
1. Upgrade to gradle 7.4.2, use `maven-publish` gradle plugin rather than deprecated `maven` (to get build working for me on jdk11)
2. Change the `EventProcessor.compose()` function to be "additive" rather than "selective" - ie henceforth if there is more than one function capable of processing a particular event, then all the functions will be executed (in order) rather than just the last. 
  a. Also change `compose()` to support more than 2 parameters

### CODE REVIEW

File: src/test/kotlin/com/cultureamp/eventsourcing/RelationalDatabaseEventStoreTest.kt [hunk 1]
```diff
@@ -100,8 +100,8 @@ class RelationalDatabaseEventStoreTest : DescribeSpec({
         it("gets the concurrency error from the sink") {
             val aggregateId = UUID.randomUUID()
             val events = listOf(
-                event(PizzaCreated(MARGHERITA, listOf(TOMATO_PASTE)), aggregateId, 1, StandardEventMetadata("unused")),
-                event(PizzaEaten(), aggregateId, 1, StandardEventMetadata("unused")),
+                event(PizzaCreated(MARGHERITA, listOf(TOMATO_PASTE)), aggregateId, 1, StandardEventMetadata(/* unused */UUID.randomUUID())),
+                event(PizzaEaten(), aggregateId, 1, StandardEventMetadata(/* unused */UUID.randomUUID())),
```
Comment (williamboxhall) [inline src/test/kotlin/com/cultureamp/eventsourcing/RelationalDatabaseEventStoreTest.kt:104]: Oh was this not compiling? 😬 this repo needs some love

**Review by williamboxhall — DISMISSED** [review]
Looks good! Keen to see if you're able to publish it without any assistance from me

**Comment by davidwheeler123** [comment]
> Looks good!

Thanks! I want to make a test for it too. 

> Keen to see if you're able to publish it without any assistance from me

I haven't been able to find the signing key, but someone else will be able to help with that. Is there a reason we don't have CI do the publish?

File: gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-base_version=0.12.0
+base_version=0.14.0
```
Comment (wfoxd) [inline gradle.properties:1]: where is 0.13?

Comment (davidwheeler123) [reply gradle.properties:1]: I couldn't work out why my change to kestrel wasn't taking effect in develop. Bumping this ver again made it work 😄

**Review by williamboxhall — APPROVED** [review]

**Pull request merged**